net/netfilter/ip6t_sockopt.c

```diff
@@ -0,0 +1,444 @@
+/****************************************************************************
+ * net/netfilter/ip6t_sockopt.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <sys/param.h>
+
+#include <nuttx/kmalloc.h>
+
+#include "netfilter/iptables.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define SWAP(a,b,t)    do { (t) = (a); (a) = (b); (b) = (t); } while (0)
```

Comment:
```suggestion
#define SWAP(a,b,t)    do { FAR void *t; (t) = (a); (a) = (b); (b) = (t); } while (0)
```

Reply:
@anchao Do you mean to change `SWAP(a,b,t)` to `SWAP(a,b)`?

Reply:
Yep, this private macro only for pointer objects, you could keep the current style if which have other references

Reply:
Done.